// samp-node-lib/build/SampEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BODY_PARTS = exports.CARMODTYPE = exports.TEXTDRAW_ALIGN = exports.BULLET_HIT_TYPE = exports.OBJECT_MATERIAL_TEXT_ALIGN = exports.OBJECT_MATERIAL_SIZE = exports.VEHICLE_MODEL_INFO = exports.WEAPON = exports.WEAPONSTATE = exports.WEAPONSKILL = exports.FIGHT_STYLE = exports.PLAYER_STATE = exports.SPECIAL_ACTION = exports.KEY = exports.DIALOG_STYLE = void 0;
var DIALOG_STYLE;
(function (DIALOG_STYLE) {
    DIALOG_STYLE[DIALOG_STYLE["MSGBOX"] = 0] = "MSGBOX";
    DIALOG_STYLE[DIALOG_STYLE["INPUT"] = 1] = "INPUT";
    DIALOG_STYLE[DIALOG_STYLE["LIST"] = 2] = "LIST";
    DIALOG_STYLE[DIALOG_STYLE["PASSWORD"] = 3] = "PASSWORD";
    DIALOG_STYLE[DIALOG_STYLE["TABLIST"] = 4] = "TABLIST";
    DIALOG_STYLE[DIALOG_STYLE["TABLIST_HEADERS"] = 5] = "TABLIST_HEADERS";
})(DIALOG_STYLE = exports.DIALOG_STYLE || (exports.DIALOG_STYLE = {}));
var KEY;
(function (KEY) {
    KEY[KEY["ACTION"] = 1] = "ACTION";
    KEY[KEY["CROUCH"] = 2] = "CROUCH";
    KEY[KEY["FIRE"] = 4] = "FIRE";
    KEY[KEY["SPRINT"] = 8] = "SPRINT";
    KEY[KEY["SECONDARY_ATTACK"] = 16] = "SECONDARY_ATTACK";
    KEY[KEY["JUMP"] = 32] = "JUMP";
    KEY[KEY["LOOK_RIGHT"] = 64] = "LOOK_RIGHT";
    KEY[KEY["HANDBRAKE"] = 128] = "HANDBRAKE";
    KEY[KEY["LOOK_LEFT"] = 256] = "LOOK_LEFT";
    KEY[KEY["SUBMISSION"] = 512] = "SUBMISSION";
    KEY[KEY["LOOK_BEHIND"] = 512] = "LOOK_BEHIND";
    KEY[KEY["WALK"] = 1024] = "WALK";
    KEY[KEY["ANALOG_UP"] = 2048] = "ANALOG_UP";
    KEY[KEY["ANALOG_DOWN"] = 4096] = "ANALOG_DOWN";
    KEY[KEY["ANALOG_RIGHT"] = 16384] = "ANALOG_RIGHT";
    KEY[KEY["ANALOG_LEFT"] = 8192] = "ANALOG_LEFT";
    KEY[KEY["UP"] = -128] = "UP";
    KEY[KEY["DOWN"] = 128] = "DOWN";
    KEY[KEY["LEFT"] = -128] = "LEFT";
    KEY[KEY["RIGHT"] = 128] = "RIGHT";
})(KEY = exports.KEY || (exports.KEY = {}));
var SPECIAL_ACTION;
(function (SPECIAL_ACTION) {
    SPECIAL_ACTION[SPECIAL_ACTION["NONE"] = 0] = "NONE";
    SPECIAL_ACTION[SPECIAL_ACTION["DUCK"] = 1] = "DUCK";
    SPECIAL_ACTION[SPECIAL_ACTION["USEJETPACK"] = 2] = "USEJETPACK";
    SPECIAL_ACTION[SPECIAL_ACTION["ENTER_VEHICLE"] = 3] = "ENTER_VEHICLE";
    SPECIAL_ACTION[SPECIAL_ACTION["EXIT_VEHICLE"] = 4] = "EXIT_VEHICLE";
    SPECIAL_ACTION[SPECIAL_ACTION["DANCE1"] = 5] = "DANCE1";
    SPECIAL_ACTION[SPECIAL_ACTION["DANCE2"] = 6] = "DANCE2";
    SPECIAL_ACTION[SPECIAL_ACTION["DANCE3"] = 7] = "DANCE3";
    SPECIAL_ACTION[SPECIAL_ACTION["DANCE4"] = 8] = "DANCE4";
    SPECIAL_ACTION[SPECIAL_ACTION["HANDSUP"] = 10] = "HANDSUP";
    SPECIAL_ACTION[SPECIAL_ACTION["USECELLPHONE"] = 11] = "USECELLPHONE";
    SPECIAL_ACTION[SPECIAL_ACTION["SITTING"] = 12] = "SITTING";
    SPECIAL_ACTION[SPECIAL_ACTION["STOPUSECELLPHONE"] = 13] = "STOPUSECELLPHONE";
    SPECIAL_ACTION[SPECIAL_ACTION["DRINK_BEER"] = 20] = "DRINK_BEER";
    SPECIAL_ACTION[SPECIAL_ACTION["SMOKE_CIGGY"] = 21] = "SMOKE_CIGGY";
    SPECIAL_ACTION[SPECIAL_ACTION["DRINK_WINE"] = 22] = "DRINK_WINE";
    SPECIAL_ACTION[SPECIAL_ACTION["DRINK_SPRUNK"] = 23] = "DRINK_SPRUNK";
    SPECIAL_ACTION[SPECIAL_ACTION["CUFFED"] = 24] = "CUFFED";
    SPECIAL_ACTION[SPECIAL_ACTION["CARRY"] = 25] = "CARRY";
    SPECIAL_ACTION[SPECIAL_ACTION["PISSING"] = 68] = "PISSING";
})(SPECIAL_ACTION = exports.SPECIAL_ACTION || (exports.SPECIAL_ACTION = {}));
var PLAYER_STATE;
(function (PLAYER_STATE) {
    PLAYER_STATE[PLAYER_STATE["NONE"] = 0] = "NONE";
    PLAYER_STATE[PLAYER_STATE["ONFOOT"] = 1] = "ONFOOT";
    PLAYER_STATE[PLAYER_STATE["DRIVER"] = 2] = "DRIVER";
    PLAYER_STATE[PLAYER_STATE["PASSENGER"] = 3] = "PASSENGER";
    PLAYER_STATE[PLAYER_STATE["EXIT_VEHICLE"] = 4] = "EXIT_VEHICLE";
    PLAYER_STATE[PLAYER_STATE["ENTER_VEHICLE_DRIVER"] = 5] = "ENTER_VEHICLE_DRIVER";
    PLAYER_STATE[PLAYER_STATE["ENTER_VEHICLE_PASSENGER"] = 6] = "ENTER_VEHICLE_PASSENGER";
    PLAYER_STATE[PLAYER_STATE["WASTED"] = 7] = "WASTED";
    PLAYER_STATE[PLAYER_STATE["SPAWNED"] = 8] = "SPAWNED";
    PLAYER_STATE[PLAYER_STATE["SPECTATING"] = 9] = "SPECTATING";
})(PLAYER_STATE = exports.PLAYER_STATE || (exports.PLAYER_STATE = {}));
var FIGHT_STYLE;
(function (FIGHT_STYLE) {
    FIGHT_STYLE[FIGHT_STYLE["NORMAL"] = 4] = "NORMAL";
    FIGHT_STYLE[FIGHT_STYLE["BOXING"] = 5] = "BOXING";
    FIGHT_STYLE[FIGHT_STYLE["KUNGFU"] = 6] = "KUNGFU";
    FIGHT_STYLE[FIGHT_STYLE["KNEEHEAD"] = 7] = "KNEEHEAD";
    FIGHT_STYLE[FIGHT_STYLE["GRABKICK"] = 15] = "GRABKICK";
    FIGHT_STYLE[FIGHT_STYLE["ELBOW"] = 16] = "ELBOW";
})(FIGHT_STYLE = exports.FIGHT_STYLE || (exports.FIGHT_STYLE = {}));
var WEAPONSKILL;
(function (WEAPONSKILL) {
    WEAPONSKILL[WEAPONSKILL["PISTOL"] = 0] = "PISTOL";
    WEAPONSKILL[WEAPONSKILL["PISTOL_SILENCED"] = 1] = "PISTOL_SILENCED";
    WEAPONSKILL[WEAPONSKILL["DESERT_EAGLE"] = 2] = "DESERT_EAGLE";
    WEAPONSKILL[WEAPONSKILL["SHOTGUN"] = 3] = "SHOTGUN";
    WEAPONSKILL[WEAPONSKILL["SAWNOFF_SHOTGUN"] = 4] = "SAWNOFF_SHOTGUN";
    WEAPONSKILL[WEAPONSKILL["SPAS12_SHOTGUN"] = 5] = "SPAS12_SHOTGUN";
    WEAPONSKILL[WEAPONSKILL["MICRO_UZI"] = 6] = "MICRO_UZI";
    WEAPONSKILL[WEAPONSKILL["MP5"] = 7] = "MP5";
    WEAPONSKILL[WEAPONSKILL["AK47"] = 8] = "AK47";
    WEAPONSKILL[WEAPONSKILL["M4"] = 9] = "M4";
    WEAPONSKILL[WEAPONSKILL["SNIPERRIFLE"] = 10] = "SNIPERRIFLE";
})(WEAPONSKILL = exports.WEAPONSKILL || (exports.WEAPONSKILL = {}));
var WEAPONSTATE;
(function (WEAPONSTATE) {
    WEAPONSTATE[WEAPONSTATE["UNKNOWN"] = -1] = "UNKNOWN";
    WEAPONSTATE[WEAPONSTATE["NO_BULLETS"] = 0] = "NO_BULLETS";
    WEAPONSTATE[WEAPONSTATE["LAST_BULLET"] = 1] = "LAST_BULLET";
    WEAPONSTATE[WEAPONSTATE["MORE_BULLETS"] = 2] = "MORE_BULLETS";
    WEAPONSTATE[WEAPONSTATE["RELOADING"] = 3] = "RELOADING";
})(WEAPONSTATE = exports.WEAPONSTATE || (exports.WEAPONSTATE = {}));
var WEAPON;
(function (WEAPON) {
    WEAPON[WEAPON["BRASSKNUCKLE"] = 1] = "BRASSKNUCKLE";
    WEAPON[WEAPON["GOLFCLUB"] = 2] = "GOLFCLUB";
    WEAPON[WEAPON["NITESTICK"] = 3] = "NITESTICK";
    WEAPON[WEAPON["KNIFE"] = 4] = "KNIFE";
    WEAPON[WEAPON["BAT"] = 5] = "BAT";
    WEAPON[WEAPON["SHOVEL"] = 6] = "SHOVEL";
    WEAPON[WEAPON["POOLSTICK"] = 7] = "POOLSTICK";
    WEAPON[WEAPON["KATANA"] = 8] = "KATANA";
    WEAPON[WEAPON["CHAINSAW"] = 9] = "CHAINSAW";
    WEAPON[WEAPON["DILDO"] = 10] = "DILDO";
    WEAPON[WEAPON["DILDO2"] = 11] = "DILDO2";
    WEAPON[WEAPON["VIBRATOR"] = 12] = "VIBRATOR";
    WEAPON[WEAPON["VIBRATOR2"] = 13] = "VIBRATOR2";
    WEAPON[WEAPON["FLOWER"] = 14] = "FLOWER";
    WEAPON[WEAPON["CANE"] = 15] = "CANE";
    WEAPON[WEAPON["GRENADE"] = 16] = "GRENADE";
    WEAPON[WEAPON["TEARGAS"] = 17] = "TEARGAS";
    WEAPON[WEAPON["MOLTOV"] = 18] = "MOLTOV";
    WEAPON[WEAPON["COLT45"] = 22] = "COLT45";
    WEAPON[WEAPON["SILENCED"] = 23] = "SILENCED";
    WEAPON[WEAPON["DEAGLE"] = 24] = "DEAGLE";
    WEAPON[WEAPON["SHOTGUN"] = 25] = "SHOTGUN";
    WEAPON[WEAPON["SAWEDOFF"] = 26] = "SAWEDOFF";
    WEAPON[WEAPON["SHOTGSPA"] = 27] = "SHOTGSPA";
    WEAPON[WEAPON["UZI"] = 28] = "UZI";
    WEAPON[WEAPON["MP5"] = 29] = "MP5";
    WEAPON[WEAPON["AK47"] = 30] = "AK47";
    WEAPON[WEAPON["M4"] = 31] = "M4";
    WEAPON[WEAPON["TEC9"] = 32] = "TEC9";
    WEAPON[WEAPON["RIFLE"] = 33] = "RIFLE";
    WEAPON[WEAPON["SNIPER"] = 34] = "SNIPER";
    WEAPON[WEAPON["ROCKETLAUNCHER"] = 35] = "ROCKETLAUNCHER";
    WEAPON[WEAPON["HEATSEEKER"] = 36] = "HEATSEEKER";
    WEAPON[WEAPON["FLAMETHROWER"] = 37] = "FLAMETHROWER";
    WEAPON[WEAPON["MINIGUN"] = 38] = "MINIGUN";
    WEAPON[WEAPON["SATCHEL"] = 39] = "SATCHEL";
    WEAPON[WEAPON["BOMB"] = 40] = "BOMB";
    WEAPON[WEAPON["SPRAYCAN"] = 41] = "SPRAYCAN";
    WEAPON[WEAPON["FIREEXTINGUISHER"] = 42] = "FIREEXTINGUISHER";
    WEAPON[WEAPON["CAMERA"] = 43] = "CAMERA";
    WEAPON[WEAPON["PARACHUTE"] = 46] = "PARACHUTE";
    WEAPON[WEAPON["VEHICLE"] = 49] = "VEHICLE";
    WEAPON[WEAPON["DROWN"] = 53] = "DROWN";
    WEAPON[WEAPON["COLLISION"] = 54] = "COLLISION";
})(WEAPON = exports.WEAPON || (exports.WEAPON = {}));
var VEHICLE_MODEL_INFO;
(function (VEHICLE_MODEL_INFO) {
    VEHICLE_MODEL_INFO[VEHICLE_MODEL_INFO["SIZE"] = 1] = "SIZE";
    VEHICLE_MODEL_INFO[VEHICLE_MODEL_INFO["FRONTSEAT"] = 2] = "FRONTSEAT";
    VEHICLE_MODEL_INFO[VEHICLE_MODEL_INFO["REARSEAT"] = 3] = "REARSEAT";
    VEHICLE_MODEL_INFO[VEHICLE_MODEL_INFO["PETROLCAP"] = 4] = "PETROLCAP";
    VEHICLE_MODEL_INFO[VEHICLE_MODEL_INFO["WHEELSFRONT"] = 5] = "WHEELSFRONT";
    VEHICLE_MODEL_INFO[VEHICLE_MODEL_INFO["WHEELSREAR"] = 6] = "WHEELSREAR";
    VEHICLE_MODEL_INFO[VEHICLE_MODEL_INFO["WHEELSMID"] = 7] = "WHEELSMID";
    VEHICLE_MODEL_INFO[VEHICLE_MODEL_INFO["FRONT_BUMPER_Z"] = 8] = "FRONT_BUMPER_Z";
    VEHICLE_MODEL_INFO[VEHICLE_MODEL_INFO["REAR_BUMPER_Z"] = 9] = "REAR_BUMPER_Z";
})(VEHICLE_MODEL_INFO = exports.VEHICLE_MODEL_INFO || (exports.VEHICLE_MODEL_INFO = {}));
var OBJECT_MATERIAL_SIZE;
(function (OBJECT_MATERIAL_SIZE) {
    OBJECT_MATERIAL_SIZE[OBJECT_MATERIAL_SIZE["_32x32"] = 10] = "_32x32";
    OBJECT_MATERIAL_SIZE[OBJECT_MATERIAL_SIZE["_64x32"] = 20] = "_64x32";
    OBJECT_MATERIAL_SIZE[OBJECT_MATERIAL_SIZE["_64x64"] = 30] = "_64x64";
    OBJECT_MATERIAL_SIZE[OBJECT_MATERIAL_SIZE["_128x32"] = 40] = "_128x32";
    OBJECT_MATERIAL_SIZE[OBJECT_MATERIAL_SIZE["_128x64"] = 50] = "_128x64";
    OBJECT_MATERIAL_SIZE[OBJECT_MATERIAL_SIZE["_128x128"] = 60] = "_128x128";
    OBJECT_MATERIAL_SIZE[OBJECT_MATERIAL_SIZE["_256x32"] = 70] = "_256x32";
    OBJECT_MATERIAL_SIZE[OBJECT_MATERIAL_SIZE["_256x64"] = 80] = "_256x64";
    OBJECT_MATERIAL_SIZE[OBJECT_MATERIAL_SIZE["_256x128"] = 90] = "_256x128";
    OBJECT_MATERIAL_SIZE[OBJECT_MATERIAL_SIZE["_256x256"] = 100] = "_256x256";
    OBJECT_MATERIAL_SIZE[OBJECT_MATERIAL_SIZE["_512x64"] = 110] = "_512x64";
    OBJECT_MATERIAL_SIZE[OBJECT_MATERIAL_SIZE["_512x128"] = 120] = "_512x128";
    OBJECT_MATERIAL_SIZE[OBJECT_MATERIAL_SIZE["_512x256"] = 130] = "_512x256";
    OBJECT_MATERIAL_SIZE[OBJECT_MATERIAL_SIZE["_512x512"] = 140] = "_512x512";
})(OBJECT_MATERIAL_SIZE = exports.OBJECT_MATERIAL_SIZE || (exports.OBJECT_MATERIAL_SIZE = {}));
var OBJECT_MATERIAL_TEXT_ALIGN;
(function (OBJECT_MATERIAL_TEXT_ALIGN) {
    OBJECT_MATERIAL_TEXT_ALIGN[OBJECT_MATERIAL_TEXT_ALIGN["LEFT"] = 0] = "LEFT";
    OBJECT_MATERIAL_TEXT_ALIGN[OBJECT_MATERIAL_TEXT_ALIGN["CENTER"] = 1] = "CENTER";
    OBJECT_MATERIAL_TEXT_ALIGN[OBJECT_MATERIAL_TEXT_ALIGN["RIGHT"] = 2] = "RIGHT";
})(OBJECT_MATERIAL_TEXT_ALIGN = exports.OBJECT_MATERIAL_TEXT_ALIGN || (exports.OBJECT_MATERIAL_TEXT_ALIGN = {}));
var BULLET_HIT_TYPE;
(function (BULLET_HIT_TYPE) {
    BULLET_HIT_TYPE[BULLET_HIT_TYPE["NONE"] = 0] = "NONE";
    BULLET_HIT_TYPE[BULLET_HIT_TYPE["PLAYER"] = 1] = "PLAYER";
    BULLET_HIT_TYPE[BULLET_HIT_TYPE["VEHICLE"] = 2] = "VEHICLE";
    BULLET_HIT_TYPE[BULLET_HIT_TYPE["OBJECT"] = 3] = "OBJECT";
    BULLET_HIT_TYPE[BULLET_HIT_TYPE["PLAYER_OBJECT"] = 4] = "PLAYER_OBJECT";
})(BULLET_HIT_TYPE = exports.BULLET_HIT_TYPE || (exports.BULLET_HIT_TYPE = {}));
var TEXTDRAW_ALIGN;
(function (TEXTDRAW_ALIGN) {
    TEXTDRAW_ALIGN[TEXTDRAW_ALIGN["LEFT"] = 1] = "LEFT";
    TEXTDRAW_ALIGN[TEXTDRAW_ALIGN["CENTER"] = 2] = "CENTER";
    TEXTDRAW_ALIGN[TEXTDRAW_ALIGN["RIGHT"] = 3] = "RIGHT";
})(TEXTDRAW_ALIGN = exports.TEXTDRAW_ALIGN || (exports.TEXTDRAW_ALIGN = {}));
var CARMODTYPE;
(function (CARMODTYPE) {
    CARMODTYPE[CARMODTYPE["SPOILER"] = 0] = "SPOILER";
    CARMODTYPE[CARMODTYPE["HOOD"] = 1] = "HOOD";
    CARMODTYPE[CARMODTYPE["ROOF"] = 2] = "ROOF";
    CARMODTYPE[CARMODTYPE["SIDESKIRT"] = 3] = "SIDESKIRT";
    CARMODTYPE[CARMODTYPE["LAMPS"] = 4] = "LAMPS";
    CARMODTYPE[CARMODTYPE["NITRO"] = 5] = "NITRO";
    CARMODTYPE[CARMODTYPE["EXHAUST"] = 6] = "EXHAUST";
    CARMODTYPE[CARMODTYPE["WHEELS"] = 7] = "WHEELS";
    CARMODTYPE[CARMODTYPE["STEREO"] = 8] = "STEREO";
    CARMODTYPE[CARMODTYPE["HYDRAULICS"] = 9] = "HYDRAULICS";
    CARMODTYPE[CARMODTYPE["FRONT_BUMPER"] = 10] = "FRONT_BUMPER";
    CARMODTYPE[CARMODTYPE["REAR_BUMPER"] = 11] = "REAR_BUMPER";
    CARMODTYPE[CARMODTYPE["VENT_RIGHT"] = 12] = "VENT_RIGHT";
    CARMODTYPE[CARMODTYPE["VENT_LEFT"] = 13] = "VENT_LEFT";
})(CARMODTYPE = exports.CARMODTYPE || (exports.CARMODTYPE = {}));
var BODY_PARTS;
(function (BODY_PARTS) {
    BODY_PARTS[BODY_PARTS["TORSO"] = 3] = "TORSO";
    BODY_PARTS[BODY_PARTS["GROIN"] = 4] = "GROIN";
    BODY_PARTS[BODY_PARTS["LEFT_ARM"] = 5] = "LEFT_ARM";
    BODY_PARTS[BODY_PARTS["RIGHT_ARM"] = 6] = "RIGHT_ARM";
    BODY_PARTS[BODY_PARTS["LEFT_LEG"] = 7] = "LEFT_LEG";
    BODY_PARTS[BODY_PARTS["RIGHT_LEG"] = 8] = "RIGHT_LEG";
    BODY_PARTS[BODY_PARTS["HEAD"] = 9] = "HEAD";
})(BODY_PARTS = exports.BODY_PARTS || (exports.BODY_PARTS = {}));
